common/rya.api/src/main/java/org/apache/rya/api/client/RyaClient.java

```diff
@@ -53,37 +55,37 @@
      */
     public RyaClient(
             final Install install,
-            final CreatePCJ createPcj,
-            final DeletePCJ deletePcj,
-            final CreatePeriodicPCJ createPeriodicPcj,
-            final DeletePeriodicPCJ deletePeriodicPcj,
-            final ListIncrementalQueries listIncrementalQueries,
-            final BatchUpdatePCJ batchUpdatePcj,
+            final Optional<CreatePCJ> createPcj,
+            final Optional<DeletePCJ> deletePcj,
+            final Optional<CreatePeriodicPCJ> createPeriodicPcj,
+            final Optional<DeletePeriodicPCJ> deletePeriodicPcj,
+            final Optional<ListIncrementalQueries> listIncrementalQueries,
+            final Optional<BatchUpdatePCJ> batchUpdatePcj,
             final GetInstanceDetails getInstanceDetails,
             final InstanceExists instanceExists,
             final ListInstances listInstances,
-            final AddUser addUser,
-            final RemoveUser removeUser,
+            final Optional<AddUser> addUser,
+            final Optional<RemoveUser> removeUser,
             final Uninstall uninstall,
             final LoadStatements loadStatements,
             final LoadStatementsFile loadStatementsFile,
             final ExecuteSparqlQuery executeSparqlQuery) {
         this.install = requireNonNull(install);
         this.createPcj = requireNonNull(createPcj);
         this.deletePcj = requireNonNull(deletePcj);
-        this.createPeriodicPcj = createPeriodicPcj;
-        this.deletePeriodicPcj = deletePeriodicPcj;
-        this.listIncrementalQueries = listIncrementalQueries;
+        this.createPeriodicPcj = requireNonNull(createPeriodicPcj);
+        this.deletePeriodicPcj = requireNonNull(deletePeriodicPcj);
+        this.listIncrementalQueries = requireNonNull(listIncrementalQueries);
         this.bactchUpdatePCJ = requireNonNull(batchUpdatePcj);
         this.getInstanceDetails = requireNonNull(getInstanceDetails);
         this.instanceExists = requireNonNull(instanceExists);
         this.listInstances = requireNonNull(listInstances);
         this.addUser = requireNonNull(addUser);
         this.removeUser = requireNonNull(removeUser);
         this.uninstall = requireNonNull(uninstall);
-        this.loadStatements = requireNonNull(loadStatements);
```

Comment:
Why did the null checks get removed?

Reply:
They shouldn't have been. Adding them back in.